### ISSUE TITLE
Error: Uncaught (in promise): cancelled at Refresher.stop()

### ISSUE THREAD

**Issue by bpetrov98** [firebase/firebase-js-sdk#7805]
### Operating System

Windows 11

### Browser Version

Chrome Version 119.0.6045.160 (Official Build) (64-bit)

### Firebase SDK Version

9.17.2

### Firebase SDK Product:

AppCheck

### Describe your project's tooling

Angular 15.0.0

### Describe the problem

Sometimes we get an uncaught promise error, which we can't find a way to catch and we get an email every time it happens on an user device. I also encountered this error in production, but it seems that the application works fine the whole time. If I don't refresh the page, the uncaught promise error won't stop showing and it spams our emails, because we send an email every time there is an error in our application.

### Steps and code to reproduce issue

We actually don't know how to reproduce this issue, because it happens randomly, so any suggestions or fixes will be appreciated.
Here is the code for our Firebase Service:
```TSX
@Injectable({ providedIn: 'root' })
export class FirebaseBrowser {
  public db!: Firestore;
  public storage!: FirebaseStorage;

  constructor() {
    if (typeof window === 'undefined') return;

    let app;
    try {
      app = initializeApp(environment.firebaseConfig);
      this.db = getFirestore(app);
      this.storage = getStorage(app);
    } catch (error) {
      console.error(error);
      throw new Error("Could't initialize firebase App");
    }

    if (!app || !this.db || !this.storage) throw new Error('Firebase initialize app');

    if (environment.ANGULAR_APP_APP_CHECK_DEBUG_TOKEN && environment.ANGULAR_APP_APP_CHECK_DEBUG_TOKEN.length) {
      self.FIREBASE_APPCHECK_DEBUG_TOKEN = environment.ANGULAR_APP_APP_CHECK_DEBUG_TOKEN;
      // this.connectToEmulators(this.store);
    }

    try {
      initializeAppCheck(app, {
        provider: new ReCaptchaV3Provider(environment.recaptcha.siteKey),
        isTokenAutoRefreshEnabled: true
      });
    } catch (error) {
      console.error(error)
    }
  }

  // ** FIRESTORE **
  public firestoreSubscription = (collection: string, documentId: string, callback: (snapshot: any) => void) => {
    try {
      return onSnapshot(doc(this.db, 'configurations', environment.lang, collection, documentId), callback, (error) => {
        console.error('Firebase Subscription Error', error);
      });
    } catch (error) {
      console.error(error);
      throw new Error('Firebase Subscription Error');
    }
  }
}
```

**Comment by DellaBitta** [comment]
Hi @bpetrov98,

Do you have any details on the error at all (stack trace, etc) so that we can narrow down where the issue might be? Thanks!

**Comment by bpetrov98** [comment]
Hi @DellaBitta,

I can provide you with the error stack trace in production, where I've replaced our domain with **example.com**.
Here is the error stack trace:

`
Error: Uncaught (in promise): cancelled at z (https://example.com/polyfills.cbf2e8875948a0c9.js:1:15977) at https://example.com/polyfills.cbf2e8875948a0c9.js:1:15067 at Kn.reject (https://example.com/polyfills.cbf2e8875948a0c9.js:1:15173) at Hn.stop (https://example.com/390.a68c9bab1fcf9cc9.js:1:7984) at https://example.com/390.a68c9bab1fcf9cc9.js:1:8106 at Generator.next (<anonymous>) at a (https://example.com/main.e261a3b519e48e56.js:1:394275) at j (https://example.com/main.e261a3b519e48e56.js:1:394477) at https://example.com/main.e261a3b519e48e56.js:1:394536 at new e (https://example.com/polyfills.cbf2e8875948a0c9.js:1:18365)
`

**Comment by DellaBitta** [comment]
Hi @bpetrov98,

Do you happen to have an unobfuscated stack trace that points to the error within Firebase? I think we found a potential scenario that might cause this error but it would be helpful to know for sure.

Additionally, given our investigation it's our assumption that this error doesn't actually cause any negative behavior (other than spamming you with emails) but it would be helpful if you could confirm that. Do you know if your app continues to operate as expected after the error?

Thanks!

**Comment by bpetrov98** [comment]
Hi again @DellaBitta,

We actually don't have any other stack trace, other than the one provided above. Thankfully, this error doesn't actually cause any negative behaviour. We just want to understand why and how it happens and try to prevent it in the future. We've used many catch blocks to try and catch the error, but unfortunately we couldn't catch it.

**Comment by DellaBitta** [comment]
Ok, I'll see if we can attempt to sort this out internally to the SDK.

**Comment by DellaBitta** [comment]
Hi @bpetrov98,

We've identified that the cancelled promise is part of standard App Check SDK behavior, but I just merged a PR that should suppress the propagation of an error message to the console logs. The change will be in our next release. I'll update this issue when it's available. Thanks for the report!

**Comment by bpetrov98** [comment]
Thank you for your assistance and quick response, @DellaBitta :)

**Comment by bpetrov98** [comment]
We can't really reproduce the issue. The track trace is exactly the same and sometimes randomly while sitting on a page it starts happening. It's just random, but I'm somehow sure that it comes from the Proactive Refresh of App Check. I know that it's expected behaviour, but I find it wrong to get errors like this.

**Comment by DellaBitta** [comment]
Hi @bpetrov98,

I just wanted to confirm that the user encountering the problem has been using the latest and greatest version of the Firebase SDK (`10.7.1`) and it's not a PWA that required a rollout for the update to take effect.

I ask because our previous attempt at fixing this problem should have prevented this issue from being encountered again, but that's based on our limited knowledge about what is actually going wrong.

We recently reviewed this issue again with the whole engineering team. I'm sorry to say that we've unsure what else could be triggering this exception given the information that has been provided.

If you can provide any more information about how to reproduce this issue then we'd definitely look into this further, but unfortunately, as of now we're not certain what's causing it, and why the attempted fix didn't work.

**Comment by bpetrov98** [comment]
Hi, @DellaBitta,
I just wanted to say that we no more experience this issue. What we did was update to the version v10.7.1. At first, we've experienced the error a couple of time and suddenly it stopped throwing the error. So, it's been a couple of weeks without receiving emails about the uncaught promise error. We can safely say that this issue is resolved! Thank you and the team for helping me out :)
